easydigitaldownloads/easy-digital-downloads
# Update recount tools to use new tables & queries #7910

Fixes #7910

## Proposed Changes:

### `EDD_Tools_Recount_All_Stats` and `EDD_Tools_Recount_Download_Stats`

1. Classes have been rewritten to use a new `edd_get_order_items()` query.
2. `edd_recount_download_stats_args` and `edd_recount_download_stats_total_args` filters have been deprecated. I chose to deprecate them because like 99% of the arguments are no longer applicable at all, but we could also choose to keep them so that the new queries can still be filtered. :shrug: 

### `EDD_Tools_Recount_Customer_Stats` and `EDD_Tools_Recount_Single_Customer_Stats`

1. Recount processes largely weren't working due to `post__not_in` no longer working with payment queries. This has been fixed separately here: https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7846#issuecomment-629236644 but I took the opportunity to rewrite this to use a new query instead. We no longer need to do the two separate queries (attached and unattached) because those errors should have been corrected during the migration. Orders should now all be associated with the correct `customer_id`.
2. Note: still using `edd_get_payment()` in the recount processes due to filters being in place that take in an `EDD_Payment` object.
3. Now enqueuing the `edd-admin-tools-export` JavaScript on the Customer Details page. This is due to that screen utilizing the exporter script to recount a single customer.

### General

1. Updated some incorrect PHP docs.
2. Corrected a few items that were incorrectly passing in a second parameter to `get_stored_data()` although the function only takes one parameter. Example: `$items = $this->get_stored_data( 'edd_reset_tables_to_truncate', false );` => `$items = $this->get_stored_data( 'edd_reset_tables_to_truncate' );`

## Testing

All of the tools in Downloads > Tools > General should be tested. A good way to do this:

1. Already have a tab open that shows the current, accurate stats. Keep this open.
2. Switch to `release/3.0` branch and run the tool.
3. Open a new window and navigate to the same page from step 1. Check out the broken stats.
4. Switch to this branch, `issue/7910` and run the tool again.
5. Refresh your window from step 3 (or open yet another) and check that the data has been updated and matches the data you had originally in step/window 1. (Assuming your data was accurate to begin with. :P )

To test the individual customer tool you need to go to Downloads > Customers, click through to a customer, click "Tools", and click `Recount Stats`.
